src/autotype/AutoTypeMatchView.cpp

```diff
@@ -78,21 +79,43 @@ void AutoTypeMatchView::keyPressEvent(QKeyEvent* event)
     }
 }
 
-void AutoTypeMatchView::setMatchList(const QList<AutoTypeMatch>& matches, bool selectFirst)
+void AutoTypeMatchView::setMatchList(const QList<AutoTypeMatch>& matches)
 {
     m_model->setMatchList(matches);
     m_sortModel->setFilterWildcard({});
 
     horizontalHeader()->resizeSections(QHeaderView::ResizeToContents);
 
-    if (selectFirst) {
-        selectionModel()->setCurrentIndex(m_sortModel->index(0, 0),
+    selectionModel()->clear();
+    emit currentMatchChanged(currentMatch());
+}
+
+void AutoTypeMatchView::selectFirstMatch()
+{
+    selectionModel()->setCurrentIndex(m_sortModel->index(0, 0),
+                                      QItemSelectionModel::ClearAndSelect | QItemSelectionModel::Rows);
+    emit currentMatchChanged(currentMatch());
+}
+
+bool AutoTypeMatchView::selectMatch(const AutoTypeMatch& match)
+{
+    int selectedIndex = -1;
+    auto matches = m_model->matchList();
+    for (int i = 0; i < matches->size(); i++) {
+        const auto& currentMatch = matches->at(i);
+        if (currentMatch.first == match.first || (selectedIndex == -1 && currentMatch.first == match.first)) {
+            selectedIndex = i;
+        }
+    }
```

Comment:
As mentioned above, this should be within the model itself since it controls the list of matches.

Reply:
Moved to model.